pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -404,6 +404,10 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[string]b
 		if isClusterLocal && (cb.proxy.Metadata.ClusterID != instance.Endpoint.Locality.ClusterID) {
 			continue
 		}
+		// TODO(nmittler): Consider merging discoverability policy with cluster-local
+		if !cb.push.IsDiscoverableFromProxy(service, instance.Endpoint, cb.proxy) {
```

Comment:
To address my previous comments, I would use a 'if featureEnabled ' - and have a local/private method here, until we optimize.

Reply:
The feature enabled flag is used at startup to select the appropriate policy. By default, the `AlwaysDiscoverable` policy is used.